File: bddtests/src/main/kotlin/mocking/tpp/models/AuthenticateReply.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "AuthenticateReply")
@XmlAccessorType(XmlAccessType.FIELD)
data class AuthenticateReply(
        @XmlAttribute var patientId: String = "default patientId",
        @XmlAttribute var onlineUserId: String = "default onlineUserId",
        @XmlAttribute var uuid: String = "default uuid",
        @field:XmlElement(name = "User") var user: User = User(
                Person(
                    "default patientId",
                    "default dateOfBirth",
                    "default gender",
                    NationalId(
                            "default type",
                            "default value"
                    ),
                    PersonName(
                            "default name"
                    )
                )
        ),
        @field:XmlElement(name= "Person") var person:  MutableCollection<Person> = mutableListOf(
                Person("default patientId",
                        "default dateOfBirth",
                        "default gender",
                        NationalId(
                                "default type",
                                "default value"
                        ),
                        PersonName(
                                "default name"
                        )
                )
        )
)

File: xamarinintegrationtests/NHSOnline.HttpMocks/Tpp/Models/AuthenticateReply.cs
using System;
using System.Collections.ObjectModel;
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Tpp.Models
{
    [Serializable]
    public class AuthenticateReply
    {
        [XmlAttribute("patientId")]
        public string? PatientId { get; set; }

        [XmlAttribute("onlineUserId")]
        public string? OnlineUserId { get; set; }

        public User? User { get; set; }

        [XmlAttribute("uuid")]
        public Guid Uuid { get; set; }

        [XmlIgnore]
        public string? Suid { get; set; }

        [XmlElement("Person")]
        public Collection<Person> People { get; } = new Collection<Person>();

        public Registration? Registration { get; set; }
    }
}